tests/callbacks/test_early_stopping.py

```diff
@@ -49,9 +48,9 @@ def on_train_start(self, trainer, pl_module):
     # ensure state is persisted properly
     checkpoint = torch.load(checkpoint_filepath)
     # the checkpoint saves "epoch + 1"
-    early_stop_callback_state = early_stop_callback.saved_states[checkpoint['epoch'] - 1]
+    early_stop_callback_state = early_stop_callback.saved_states[checkpoint["epoch"] - 1]
     assert 4 == len(early_stop_callback.saved_states)
-    assert checkpoint['early_stop_callback_state_dict'] == early_stop_callback_state
+    assert checkpoint["callbacks"][type(early_stop_callback)] == early_stop_callback_state
```

Comment:
so essentially the upgrade script should replace the old key `checkpoint['early_stop_callback_state_dict']` with `checkpoint["callbacks"][type(early_stop_callback)] `that should be all, right? The only thing we don't know is the type (if user had overridden it), but I guess we have to assume it was the EarlyStoppingCallback.

Reply:
Yeah, we'll need to do the same thing for model checkpointing as well (and that case is a little more complex). 

`checkpoint['checkpoint_callback_best']` -> `checkpoint["callbacks"][type(model_checkpoint_callback)]['best_model_score']` (old format)

`checkpoint['checkpoint_callback_best_model_score']` -> `checkpoint["callbacks"][type(model_checkpoint_callback)]['best_model_score']` (newer format)

`checkpoint['checkpoint_callback_best_model_path']` -> `checkpoint["callbacks"][type(model_checkpoint_callback)]['best_model_path']`

Reply:
sounds good. is it fine if we run the conversion automatically and print a info/warning message? I think we don't even have to write it back to disk, only do it internally, online, and new checkpoints get saved in the new format.